antarest/study/business/binding_constraint_management.py

```diff
@@ -178,167 +391,329 @@ def parse_constraint(key: str, value: str, char: str, new_config: BindingConstra
         return False
 
     @staticmethod
-    def process_constraint(constraint_value: Dict[str, Any], version: int) -> BindingConstraintConfigType:
-        args = {
-            "id": constraint_value["id"],
-            "name": constraint_value["name"],
-            "enabled": constraint_value["enabled"],
-            "time_step": constraint_value["type"],
-            "operator": constraint_value["operator"],
-            "comments": constraint_value.get("comments", None),
-            "filter_year_by_year": constraint_value.get("filter-year-by-year", ""),
-            "filter_synthesis": constraint_value.get("filter-synthesis", ""),
-            "constraints": None,
+    def constraint_model_adapter(constraint: ConstraintInput, version: int) -> ConstraintOutput:
+        """
+        Adapts a constraint configuration to the appropriate version-specific format.
+
+        Parameters:
+        - constraint: A dictionary or model representing the constraint to be adapted.
+                    This can either be a dictionary coming from client input or an existing
+                    model that needs reformatting.
+        - version: An integer indicating the target version of the study configuration. This is used to
+                determine which model class to instantiate and which default values to apply.
+
+        Returns:
+        - A new instance of either `ConstraintOutputBase` or `ConstraintOutput870`,
+        populated with the adapted values from the input constraint, and conforming to the
+        structure expected by the specified version.
+
+        Note:
+        This method is crucial for ensuring backward compatibility and future-proofing the application
+        as it evolves. It allows client-side data to be accurately represented within the config and
+        ensures data integrity when storing or retrieving constraint configurations from the database.
+        """
+
+        constraint_output = {
+            "id": constraint["id"],
+            "name": constraint["name"],
+            "enabled": constraint.get("enabled", True),
+            "time_step": constraint.get("type", BindingConstraintFrequency.HOURLY),
+            "operator": constraint.get("operator", BindingConstraintOperator.EQUAL),
+            "comments": constraint.get("comments", ""),
+            "filter_year_by_year": constraint.get("filter_year_by_year", ""),
```

Comment:
Are you aware that `filter_year_by_year` and `filter_synthesis` are only available for study v8.3 or above?

See File format changes [v8.3.2 — Marginal price of a binding constraint](https://antares-simulator.readthedocs.io/en/latest/reference-guide/13-file-format/#marginal-price-of-a-binding-constraint)

Reply:
This is handled on the front end, but if it causes problems for the simulator by adding data through the API directly, we should handle it (we didn't before).